pkg/pipeline/CdHandler.go

```diff
@@ -973,6 +991,75 @@ func (impl *CdHandlerImpl) getLogsFromRepository(pipelineId int, cdWorkflow *pip
 	logReader := bufio.NewReader(oldLogsStream)
 	return logReader, cleanUp, err
 }
+func assignNewBlobStorageConfigInCdLogRequest(cdLogRequest *BuildLogRequest, cmConfig *bean2.CmBlobStorageConfig, secretConfig *bean2.SecretBlobStorageConfig) BuildLogRequest {
+	cdLogRequest.CloudProvider = cmConfig.CloudProvider
+	cdLogRequest.AzureBlobConfig.AccountName = cmConfig.AzureAccountName
+	cdLogRequest.AzureBlobConfig.AccountKey = decodeSecretKey(secretConfig.AzureAccountKey)
+	cdLogRequest.AzureBlobConfig.BlobContainerName = cmConfig.AzureBlobContainerCiLog
+
+	cdLogRequest.GcpBlobBaseConfig.CredentialFileJsonData = decodeSecretKey(secretConfig.GcpBlobStorageCredentialJson)
+	cdLogRequest.GcpBlobBaseConfig.BucketName = cmConfig.CdDefaultBuildLogsBucket
+
+	cdLogRequest.AwsS3BaseConfig.AccessKey = cmConfig.S3AccessKey
+	cdLogRequest.AwsS3BaseConfig.EndpointUrl = cmConfig.S3Endpoint
+	cdLogRequest.AwsS3BaseConfig.Passkey = decodeSecretKey(secretConfig.S3SecretKey)
+	isEndpointInSecure, _ := strconv.ParseBool(cmConfig.S3EndpointInsecure)
+	cdLogRequest.AwsS3BaseConfig.IsInSecure = isEndpointInSecure
+	cdLogRequest.AwsS3BaseConfig.BucketName = cmConfig.CdDefaultBuildLogsBucket
+	cdLogRequest.AwsS3BaseConfig.Region = cmConfig.CdDefaultCdLogsBucketRegion
+	s3BucketVersioned, _ := strconv.ParseBool(cmConfig.S3BucketVersioned)
+	cdLogRequest.AwsS3BaseConfig.VersioningEnabled = s3BucketVersioned
+
+	return *cdLogRequest
+}
+
+func (impl *CdHandlerImpl) FetchCmAndSecretBlobConfigFromExternalCluster(clusterConfig *k8s.ClusterConfig, namespace string) (*bean2.CmBlobStorageConfig, *bean2.SecretBlobStorageConfig, error) {
+	cmConfig := &bean2.CmBlobStorageConfig{}
+	secretConfig := &bean2.SecretBlobStorageConfig{}
+	_, _, kubeClient, err := impl.k8sUtil.GetK8sConfigAndClients(clusterConfig)
+	if err != nil {
+		impl.Logger.Errorw("FetchCmAndSecretBlobConfigFromExternalCluster, error in getting kubeClient by cluster config", "err", err)
+		return cmConfig, secretConfig, err
+	}
+	cm, err := kubeClient.CoreV1().ConfigMaps(namespace).Get(context.Background(), impl.config.ExtBlobStorageCmName, v12.GetOptions{})
+	if err != nil && !errors2.IsNotFound(err) {
+		impl.Logger.Errorw("error in getting config map in external cluster", "err", err, "blobStorageCmName", impl.config.ExtBlobStorageCmName, "clusterName", clusterConfig.ClusterName)
+		return cmConfig, secretConfig, err
+	}
+	if errors2.IsNotFound(err) {
+		cmNotFoundErr := fmt.Sprintf("config map: %s not found in namespace: %s", impl.config.ExtBlobStorageCmName, namespace)
+		return cmConfig, secretConfig, errors.New(cmNotFoundErr)
+	}
```

Comment:
what's the benefit of handling different errors separately here?

Reply:
no benefit as such, wanted to send personalised messaging that this cm is not found in this particular namespace